### PR TITLE
Signup: Alias anonUserId to WPCOM user id when user is created

### PR DESCRIPTION
**Pull request by bisko**
This PR aims to properly connect the `anonUserId` and the WPCOM user id. 

Context: p3Ex-24k-p2

To test:

1. Checkout branch
2. Go through Signup without adding anything into Cart (domain or plan)
3. Reach User step
4. Fill the user step
5. Submit
6. Verify the `_aliasUser ` event has been properly sent to the analytics backend

### CODE REVIEW

File: client/lib/analytics/index.js [hunk 1]
```diff
@@ -382,17 +382,20 @@ const analytics = {
 			s.parentNode.insertBefore( wa, s );
 		}
 	},
-
-	identifyUser: function() {
+
```
Comment (michaeldcain) [inline client/lib/analytics/index.js]: `L385: Trailing spaces not allowed`

File: client/lib/signup/step-actions.js [hunk 2]
```diff
@@ -239,7 +239,7 @@ module.exports = {
 	createAccount( callback, dependencies, { userData, flowName, queryArgs, service, token }, reduxStore ) {
 		const surveyVertical = getSurveyVertical( reduxStore.getState() ).trim();
 		const surveySiteType = getSurveySiteType( reduxStore.getState() ).trim();
-
+
```
Comment (michaeldcain) [inline client/lib/signup/step-actions.js]: I bet you can guess ;)

**Comment by xyu** [comment]
This looks good to me but I don't actually know how the user object/state and signup actions are handled in Calypso. Perhaps someone with more Calypso experience can take a look? @rralian can you take a look or suggest someone?

**Comment by michaeldcain** [comment]
>but I don't actually know how the user object/state and signup actions are handled in Calypso.

I'm happy to take a look at it from here if you don't see any data blockers.

File: client/lib/signup/step-actions.js [hunk 3]
```diff
@@ -253,27 +253,49 @@ module.exports = {
...
 
-				callback( errors, assign( {}, { username: userData.username }, bearerToken ) );
-			} );
-		}
+				// Force fetch the user details
+				user.fetch();
+			}
+
+			callback( errors, assign( {}, { username: userData.username }, bearerToken ) );
+		} );
```
Comment (michaeldcain) [inline client/lib/signup/step-actions.js]: It looks like we lost an indentation level at L256, and thus lost the closing bracket for the conditional on L54.

File: client/signup/main.jsx [hunk 4]
```diff
@@ -238,7 +238,15 @@ const Signup = React.createClass( {
 		if ( userIsLoggedIn ) {
 			// deferred in case the user is logged in and the redirect triggers a dispatch
 			defer( function() {
-				page( destination );
+				/**
+				 * The destination URL is not always inside Calypso.
+				 * For these cases make sure that a proper redirect is issued.
+				 */
+				if ( /^https?:\/\// ) {
```
Comment (michaeldcain) [inline client/signup/main.jsx]: This works for me, and I can't think of a way that it'd currently mess up. No SSR to worry about. Might be worth adding a note to the `destination` explanation in the Signup readme, but other than that, :+1:.

**Pull request merged**